datastream-kafka-connector/src/main/java/com/linkedin/datastream/connectors/kafka/AbstractKafkaBasedConnectorTask.java

```diff
@@ -774,6 +780,22 @@ protected void preConsumerPollHook() {
     }
   }
 
+  /**
+   * Pre commit hook for all operations that need to be performed before committing offsets.
+   */
+  protected void preCommitHook() { }
+
+  /**
+   * Post commit hook for all operations that need to be performed after committing offsets.
+   * <p>
+   *     Note: A commit attempt can be declared successful even when there's an exception. This can happen when a commit
+   *     exception is thrown during task shutdown.
+   * </p>
+   * @param success Indicates whether the commit attempt was successful or not.
+   * @param exception Exception caught during a commit attempt.
+   */
+  protected void postCommitHook(boolean success, KafkaException exception) { }
```

Comment:
Since, we don't have a concrete usage for this hook in open source and we currently have identified only the need for preCommit hook, will it be okay to postpone it, till either we identify the use for it or we get any pull/feature request. This will help us in avoiding an unnecessary tech debt to keep on maintaining.